lib/client-api/src/story_store.ts

```diff
@@ -276,45 +305,43 @@ export default class StoryStore extends EventEmitter {
     );
   }
 
-  setSelection(data: Selection | undefined, error: ErrorLike): void {
-    this._selection =
-      data === undefined ? this._selection : { storyId: data.storyId, viewMode: data.viewMode };
-    this._error = error === undefined ? this._error : error;
+  clearError() {
+    this._error = null;
+  }
 
-    // Try and emit the STORY_RENDER event synchronously, but if the channel is not ready (RN),
-    // we'll try again later.
-    let isStarted = false;
-    if (this._channel) {
-      this._channel.emit(Events.STORY_RENDER);
-      isStarted = true;
-    }
+  setError = (err: ErrorLike) => {
+    this._error = err;
+    if (this._channel) this._channel.emit(Events.RENDER_CURRENT_STORY);
+  };
 
-    setTimeout(() => {
-      if (this._channel && !isStarted) {
-        this._channel.emit(Events.STORY_RENDER);
-      }
+  getError = (): ErrorLike | undefined => this._error;
 
-      // should be deprecated in future.
-      this.emit(Events.STORY_RENDER);
-    }, 1);
+  setSelection(selection: Selection): void {
+    this._selection = selection;
+
+    if (this._channel) {
+      this._channel.emit(Events.CURRENT_STORY_WAS_SET, this._selection);
+
+      // If the selection is set while configuration is in process, we are guaranteed
+      // we'll emit RENDER_CURRENT_STORY at the end of the process, so we shouldn't do it now.
+      if (!this._configuring) this._channel.emit(Events.RENDER_CURRENT_STORY);
+    }
   }
 
   getSelection = (): Selection => this._selection;
 
-  getError = (): ErrorLike | undefined => this._error;
-
   getStoriesForManager = () => {
     return this.extract({ includeDocsOnly: true });
   };
 
-  pushToManager = debounce(() => {
+  pushToManager = () => {
```

Comment:
what is the impact of removing debounce here?

Reply:
As discussed on top of ticket/in README.